library/common/main_interface.h

```diff
@@ -106,9 +106,12 @@ envoy_status_t set_preferred_network(envoy_network_t network);
  * Increment a counter with the given elements and by the given count.
  * @param engine, the engine that owns the counter.
  * @param elements, the string that identifies the counter to increment.
+ * @param tag_arr, the array of {key, value} pairs of tags.
```

Comment:
I'd create an envoy_pulse_tags (or better name) struct in order to hold both the tags and the length.